runner_test.go

```diff
@@ -367,6 +357,250 @@ func (*RunnerSuite) TestFatalErrorWhileSelfStartWorker(c *gc.C) {
 	c.Assert(err, gc.Equals, fatalStarter.startErr)
 }
 
+func (*RunnerSuite) TestWorkerWithNoWorker(c *gc.C) {
+	runner := worker.NewRunner(worker.RunnerParams{
+		IsFatal:      allFatal,
+		RestartDelay: time.Millisecond,
+	})
+	w, err := runner.Worker("id", nil)
+	c.Assert(errors.Cause(err), gc.Equals, worker.ErrNotFound)
+	c.Assert(w, gc.Equals, nil)
+}
+
+func (*RunnerSuite) TestWorkerWithWorkerImmediatelyAvailable(c *gc.C) {
+	runner := worker.NewRunner(worker.RunnerParams{
+		IsFatal:      allFatal,
+		RestartDelay: time.Millisecond,
+	})
+	starter := newTestWorkerStarter()
+	runner.StartWorker("id", starter.start)
+
+	// Wait long enough to be reasonably confident that the
+	// worker has been started and registered.
+	time.Sleep(10 * time.Millisecond)
```

Comment:
Because we test a different path through the code (verifiable with go test -cover).